File: mini-coupang/seller-service/src/main/java/com/deepblog/seller/repository/SellerRepository.java
package com.deepblog.seller.repository;

import com.deepblog.seller.entity.Seller;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface SellerRepository extends JpaRepository<Seller, Long> {

    Optional<Seller> findByEmail(String email);

    boolean existsByEmail(String email);

    boolean existsByBusinessRegistrationNo(String businessRegistrationNo);
}

File: services/mini-coupang/src/main/kotlin/com/deepblog/minicoupang/domain/seller/storage/SellerRepository.kt
package com.deepblog.minicoupang.domain.seller.storage

import com.deepblog.minicoupang.domain.seller.Seller
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface SellerRepository : JpaRepository<Seller, Long> {
    fun findByEmail(email: String): Seller?
}
